.test-infra/jenkins/job_beam_PerformanceTests_FileBasedIO_IT_HDFS.groovy

```diff
@@ -0,0 +1,124 @@
+                itClass           : 'org.apache.beam.sdk.io.text.TextIOIT',
+                bqTable           : 'beam_performance.textioit_hdfs_pkb_results',
+                prCommitStatusName: 'Java TextIO Performance Test on HDFS',
+                prTriggerPhase    : 'Run Java TextIO Performance Test HDFS',
+
+        ],
+        [
+                jobName            : 'beam_PerformanceTests_Compressed_TextIOIT_HDFS',
+                jobDescription     : 'Runs PerfKit tests for TextIOIT with GZIP compression on HDFS',
+                itClass            : 'org.apache.beam.sdk.io.text.TextIOIT',
+                bqTable            : 'beam_performance.compressed_textioit_hdfs_pkb_results',
+                prCommitStatusName : 'Java CompressedTextIO Performance Test on HDFS',
+                prTriggerPhase     : 'Run Java CompressedTextIO Performance Test HDFS',
+                extraPipelineArgs: [
+                        compressionType: 'GZIP'
+                ]
+        ],
+        [
+                jobName           : 'beam_PerformanceTests_AvroIOIT_HDFS',
+                jobDescription    : 'Runs PerfKit tests for AvroIOIT on HDFS',
+                itClass           : 'org.apache.beam.sdk.io.avro.AvroIOIT',
+                bqTable           : 'beam_performance.avroioit_hdfs_pkb_results',
+                prCommitStatusName: 'Java AvroIO Performance Test on HDFS',
+                prTriggerPhase    : 'Run Java AvroIO Performance Test HDFS',
+        ],
+        [
+                jobName           : 'beam_PerformanceTests_TFRecordIOIT_HDFS',
+                jobDescription    : 'Runs PerfKit tests for beam_PerformanceTests_TFRecordIOIT on HDFS',
+                itClass           : 'org.apache.beam.sdk.io.tfrecord.TFRecordIOIT',
+                bqTable           : 'beam_performance.tfrecordioit_hdfs_pkb_results',
+                prCommitStatusName: 'Java TFRecordIO Performance Test on HDFS',
+                prTriggerPhase    : 'Run Java TFRecordIO Performance Test HDFS',
+        ],
+]
+
+for (testConfiguration in testsConfigurations) {
+    create_filebasedio_performance_test_job(testConfiguration)
+}
+
+
+private void create_filebasedio_performance_test_job(testConfiguration) {
+
+    // This job runs the file-based IOs performance tests on PerfKit Benchmarker.
+    job(testConfiguration.jobName) {
+        description(testConfiguration.jobDescription)
+
+        // Set default Beam job properties.
+        common_job_properties.setTopLevelMainJobProperties(delegate)
+
+        // Allows triggering this build against pull requests.
+        common_job_properties.enablePhraseTriggeringFromPullRequest(
+                delegate,
+                testConfiguration.prCommitStatusName,
+                testConfiguration.prTriggerPhase)
+
+        // Run job in postcommit every 12 hours, don't trigger every push, and
+        // don't email individual committers.
+        common_job_properties.setPostCommit(
```

Comment:
We should not trigger this job till Kubernetes issues are fixed. Shall we add the test without Jenkins configs till that is figured out ?

Reply:
Sure.